velox/exec/tests/TableWriteTest.cpp

```diff
@@ -610,3 +644,111 @@ TEST_F(TableWriteTest, writeNoFile) {
   execute(plan, std::make_shared<core::QueryCtx>(executor_.get()));
   ASSERT_TRUE(fs::is_empty(outputDirectory->path));
 }
+
+TEST_F(TableWriteTest, immutablePartitionsExistingTableErrorWhenEnabled) {
+  auto input = makeVectors(rowType_, 10, 10);
+  auto outputDirectory = TempDirectoryPath::create();
+  auto plan = createInsertPlan(
+      rowType_,
+      input,
+      connector::hive::LocationHandle::TableType::kExisting,
+      outputDirectory->path);
+
+  VELOX_ASSERT_THROW(
+      AssertQueryBuilder(plan)
+          .connectorConfig(
+              kHiveConnectorId,
+              HiveConfig::kImmutablePartitions,
+              folly::to<std::string>("true"))
+          .copyResults(pool()),
+      "Unpartitioned Hive tables are immutable.");
+}
+
+TEST_F(TableWriteTest, immutablePartitionsNewTableEnabled) {
```

Comment:
This test and the previous one are exactly the same (minus the tableType). Please can you refactor this as a single function that takes the tableType as a parameter.